src/libertem_live/detectors/merlin/data.py

```diff
@@ -490,27 +490,43 @@ def get_impl(self, chunk_size=10, read_upto_frame=None):
         )
 
 
-def merlin_stream_single_threaded(backend: MerlinDataSocket, read_dtype=np.float32):
-    with backend:
-        hdr, frame_header = backend.read_headers()
-        logger.info(hdr, frame_header)
+class MerlinDataSource:
+    def __init__(self, host, port, pool_size=2):
+        self.socket = MerlinDataSocket(host=host, port=port)
+        self.pool = ReaderPool(backend=self.socket, pool_size=pool_size)
 
-        chunk_size = 10
+    def __enter__(self):
+        self.socket.__enter__()
 
-        out = backend.get_out_buffer(chunk_size, dtype=read_dtype)
-        input_buffer = backend.get_input_buffer(num_frames=chunk_size)
+    def __exit__(self, *args, **kwargs):
+        self.socket.__exit__(*args, **kwargs)
 
-        while True:
-            backend.read_multi_frames(out=out, num_frames=chunk_size, input_buffer=input_buffer)
-            yield out
+    def stream_single_threaded(self, read_dtype=np.float32, chunk_size=10):
```

Comment:
Perhaps

```suggestion
    def inline_stream(self, read_dtype=np.float32, chunk_size=10):
```

to keep it consistent with the InlineJobExecutor? :-)